# Week 1 Genome Assembly -- Feedback

1 + 1 + 1 + 1 + 0.75 + 1 + 0.75 + 0.9075 + 0.75 + 1  = 9.2 points out of 10 possible points

1. Question 1.1, 1.4 how many reads (0.5 pts each)

  * yes, (what equation did you use?) --> +1

2. Question 1.2, 1.4 simulation script(s)

  * This solution is great! Not what I was expecting, but you use a lot of numpy functions to your benefit.
  * Consider using a function that you can pass your generalized variables to rather than having repetitive code
  * --> +1

3. Question 1.2, 1.4 plotting script(s)

  * nice code! --> +1

4. Question 1.2, 1.4 histograms with overlaid Poisson distributions (0.5 pts each)

  * Fantastic plots! --> +1

5. Question 1.3, 1.4 how much of genome not sequenced/comparison to Poisson expectations (0.5 pts each, 0.25 for answer and 0.25 for code)

  * How did you find your poisson expectations? The value for 15x isn't what I would expect --> +0.75

6. Question 2 De novo assembly (0.5 pts each, 0.25 for answer and 0.25 for code)

  * number of contigs --> + 0.5
  * total length of contigs --> + 0.5

7. Question 2 De novo assembly cont (0.5 pts each, 0.25 for answer and 0.25 for code)

  * size of largest contig --> +0.5
  * contig n50 size --> +0.25, please explain your reasoning

8. whole genome alignment (0.33 pts each, 0.33/2 for answer and 0.33/2 for code)

  * average identity --> +0.33
  * length of longest alignment --> + 0.33
  * how many insertions and deletions in assembly; incorrect number of deletions --> 0.2475 (+ 0.33/4 for half answer + 0.33/2 for explanation/code)

9. decoding the insertion (0.5 pts each, 0.25 for answer and 0.25 for code)

  * position of insertion in assembly --> +0.5
  * length of novel insertion --> +0.25, incorrect length

10. decoding the insertion cont (0.5 pts each, 0.25 for answer and 0.25 for code)

  * DNA sequence of encoded message --> +0.5
  * secret message --> +0.5
