# QBB2022 - Day 1 - Lunch Exercises Submission

1. Iâ€™m excited to learn how to organize and analyze large data sets.
=======
2.
b. ~62 exons per gene. 
wc -l ~/qbb2022-answers/day1-lunch/genes.chr21.bed wc -l ~/qbb2022-answers/day1-lunch/exons.chr21.bed. 
c. I would map the exons to their corresponding genes based on the location columns. Then I would sort based on the number of exons per gene, then use wc to determine the length of the list, and then use head for half that list and then use the last value returned by head for the median.
=======  
3.
b.
 305 1
 678 2
  79 3
 377 4
 808 5
 148 6
1050 7
 156 8
 654 9
  17 10
  17 11
  30 12
  62 13
 228 14
 992 15
cut -f 4 ~/qbb2022-answers/day1-lunch/chromHMM.E116_15_coreMarks_hg38lift_stateno.chr21.bed | sort -n | uniq -c
c. I would sort by the various states and then for each state I would add up the values generated by subtracting the start position from the end position of each region in that state in order to find what state covers the largest part of the chromosome.
=======
4.
b.
 123 ACB
 112 ASW
 173 ESN
 180 GWD
 122 LWK
 128 MSL
 206 YRI
grep AFR integrated_call_samples.panel  | cut -f 2 | sort | uniq -c
c. I would perform the same command for each superpopulation replacing AFR with the abbreviation for the respective superpopulation. 